Artesian/Artesian.SDK/Factory/BidAsk.cs

```diff
@@ -0,0 +1,172 @@
+                throw new BidAskException("This MarketData has Time granularity. Use AddData(Instant time...)");
+
+            return _addBidAsk(localDate.AtMidnight(), product, value);
+        }
+        /// <summary>
+        /// BidAsk AddData
+        /// </summary>
+        /// <remarks>
+        /// Add Data on to the curve with Instant
+        /// </remarks>
+        /// <returns>AddBidAskOperationResult</returns>
+        public AddBidAskOperationResult AddData(Instant time, string product, BidAskValue value)
+        {
+            if (!_entity.OriginalGranularity.IsTimeGranularity())
+                throw new BidAskException("This MarketData has Date granularity. Use AddData(LocalDate date...)");
+
+            return _addBidAsk(time.InUtc().LocalDateTime, product, value);
+        }
+
+        private AddBidAskOperationResult _addBidAsk(LocalDateTime reportTime, string product, BidAskValue value)
+        {
+            //Relative products
+            if (Regex.IsMatch(product, @"\+\d+$"))
+                throw new NotSupportedException("Relative Products are not supported");
+
+            if (_entity.OriginalGranularity.IsTimeGranularity())
+            {
+                var period = ArtesianUtils.MapTimePeriod(_entity.OriginalGranularity);
+                if (!reportTime.IsStartOfInterval(period))
+                    throw new BidAskException("Trying to insert Report Time {0} with the wrong format to BidAsk {1}. Should be of period {2}", reportTime, _identifier, period);
+            }
+            else
+            {
+                var period = ArtesianUtils.MapDatePeriod(_entity.OriginalGranularity);
+                if (!reportTime.IsStartOfInterval(period))
+                    throw new BidAskException("Trying to insert Report Time {0} with wrong the format to BidAsk {1}. Should be of period {2}", reportTime, _identifier, period);
+            }
+
+
+            if (BidAsks.Any(row => row.ReportTime == reportTime && row.Product.Equals(product)))
+                return AddBidAskOperationResult.ProductAlreadyPresent;
+
+            BidAsks.Add(new BidAskElement(reportTime, product, value));
+            return AddBidAskOperationResult.BidAskAdded;
+        }
+
+        /// <summary>
+        /// MarketData Save
+        /// </summary>
+        /// <remarks>
+        /// Save the Data of the current MarketData
+        /// </remarks>
+        /// <param name="downloadedAt">Downloaded at</param>
+        /// <param name="deferCommandExecution">DeferCommandExecution</param>
+        /// <param name="deferDataGeneration">DeferDataGeneration</param>
+        /// <param name="keepNulls">if <see langword="false"/> nulls are ignored (server-side). That is the default behaviour.</param>
+        /// <returns></returns>
+        public async Task Save(Instant downloadedAt, bool deferCommandExecution = false, bool deferDataGeneration = true, bool keepNulls = false)
```

Comment:
I know it's aside this PR, but all Save calls miss the CTK. Could you please add it to them all?